nuxeo-services/login/nuxeo-platform-login-token-rest/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/login/tokenauth/AuthenticationTokensObject.java

```diff
@@ -65,12 +70,20 @@ protected void initialize(Object... args) {
     }
 
     @POST
+    @Produces({ MediaType.APPLICATION_JSON, MediaType.TEXT_PLAIN })
     public Response createToken(@QueryParam("application") String applicationName,
             @QueryParam("deviceId") String deviceId, @QueryParam("deviceDescription") String deviceDescription,
-            @QueryParam("permission") String permission) {
+            @QueryParam("permission") String permission, @HeaderParam("accept") String acceptType) {
         String username = getCurrentUser().getName();
         String token = service.acquireToken(username, applicationName, deviceId, deviceDescription, permission);
-        return Response.ok(token).status(Response.Status.CREATED).build();
+        String body = token;
+        MediaType cType = MediaType.TEXT_PLAIN_TYPE;
+        List<MediaType> acceptList = accepts(acceptType);
+        if (acceptList.isEmpty() || acceptList.stream().anyMatch(type -> MediaType.APPLICATION_JSON_TYPE.isCompatible(type))) {
```

Comment:
Instead of doing manual parsing/matching of the `Accept:` header, wouldn't it be simpler to have two methods, one with `@Produces(MediaType.APPLICATION_JSON)` and one with `@Produces(MediaType.TEXT_PLAIN)` to let the JAX-RS framework itself do the matching?

Reply:
Delegated matching to JAX-RS and added "detail" query parameter to return populated JSON object.